src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js

```diff
@@ -1,126 +1,74 @@
-import { dataEntryHasChanges as getDataEntryHasChanges } from '../../DataEntry/common/dataEntryHasChanges';
+    useValidatedIDsFromCache,
+} from '../../../utils/cachedDataHooks/useValidatedIDsFromCache';
+import { useCommonEnrollmentDomainData } from '../common/EnrollmentOverviewDomain';
+import { EnrollmentAddEventPageStatuses } from './EnrollmentAddEventPage.constants';
+import { LoadingMaskForPage } from '../../LoadingMasks';
 
-export const EnrollmentAddEventPage = () => {
-    const { programId, stageId, orgUnitId, teiId, enrollmentId } = useLocationQuery();
-
-    const dispatch = useDispatch();
-
-    const handleCancel = useCallback(() => {
-        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
-    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
-
-    const handleSave = useCallback(
-        (data, uid) => {
-            dispatch(updateEnrollmentEventsWithoutId(uid, data.events[0]));
-            dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
-        },
-        [dispatch, programId, orgUnitId, teiId, enrollmentId],
-    );
-
-    const handleDelete = useCallback(() => {
-        dispatch(batchActions([
-            deleteEnrollment({ enrollmentId }),
-            navigateToEnrollmentPage(programId, orgUnitId, teiId),
-        ]));
-    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
-
-    const widgetReducerName = 'enrollmentEvent-newEvent';
-
-    const dataEntryHasChanges = useSelector(state => getDataEntryHasChanges(state, widgetReducerName));
-
-    // TODO: Validate query params
-    // This includes prechecking that we got a valid program stage and move the program stage logic in this file to useEnrollmentAddEventTopBar
-    // Ticket: https://jira.dhis2.org/browse/TECH-669
-    const { program } = useProgramInfo(programId);
-    const selectedProgramStage = [...program.stages.values()].find(item => item.id === stageId);
-    const outputEffects = useWidgetDataFromStore(widgetReducerName);
-    const hideWidgets = useHideWidgetByRuleLocations(program.programRules);
+export const EnrollmentAddEventPage: ComponentType<{||}> = () => {
+    const { teiId, programId, orgUnitId, enrollmentId } = useLocationQuery();
+    const { valid: validIds, error: validatedIdsError } = useValidatedIDsFromCache({ programId, orgUnitId });
     const {
         enrollment,
         attributeValues,
         error: commonDataError,
     } = useCommonEnrollmentDomainData(teiId, enrollmentId, programId);
+    const pageIsInvalid = (validIds && !validIds?.every(({ valid }) => valid)) || commonDataError || validatedIdsError;
+    const pageStatus = useMemo(() => {
+        if (!programId || !enrollmentId || !teiId) return EnrollmentAddEventPageStatuses.MISSING_REQUIRED_VALUES;
+        if (pageIsInvalid && validIds[0]?.valid && !validIds[1]?.valid) return EnrollmentAddEventPageStatuses.ORG_UNIT_INVALID;
```

Comment:
Hey, 
I think the code would be easier to understand if `validIds[0]` and `validIds[1]` were variables with a descriptive name. Currently, you have to dig inside `useValidatedIDsFromCache` to understand what they are. Do you think naming them will improve the code readability?
Thanks!

Reply:
Yep, absolutely agree! Made some changes, hope it's better this time.